swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -141,7 +141,8 @@ class Adapter {
       // Update state in case everything succeeded
       self.state = .tunnelStarted(session: session)
     } catch let error {
-      throw AdapterError.connlibConnectError(error)
+      let msg = error as! RustString
+      throw AdapterError.connlibConnectError(msg.toString())
```

Comment:
```suggestion
      // `toString` needed to deep copy the string and avoid a possible dangling pointer
      throw AdapterError.connlibConnectError(msg.toString())
```

If I understood it correctly, a comment to prevent accidental removal would be nice